src/pages/edge/adobe-journey-optimizer/content-card-ui/iOS/tutorial/listening-content-card-events.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Listening to Content Card Events
+description: Tutorial on how to listen to content card events in your application.
+keywords:
+- Adobe Journey Optimizer
+- Guide
+- Content Card
+- Messaging
+- Customizing UI
+- Card Templates
+- Content Card Templates
+- ContentCareUI
+- Dismiss
+- Interact
+- Display
+---
+
+# Listening to Content Card Events
+
+This tutorial explains how to listen to content card events in your application.
+
+## Overview
+
+The Messaging extension provides a way to listen to events from content cards displayed in your application. The following functions can be implemented in conformance with the [ContentCardUIEventListening](../public-classes/contentcarduieventlistening.md) protocol:
+
+- `onDisplay`
+- `onDismiss`
+- `onInteract`
+
+## Implement ContentCardEventListening
+
+Complete the following steps to hear content card events:
+
+1. Conform to the [ContentCardUIEventListening](../public-classes/contentcarduieventlistening.md) protocol in your class or struct and implement the desired methods.
+1. Pass the listener to the [getContentCardsUI](../api-usage.md#getcontentcardsui) API.
+
+Below is an example implementation of `ContentCardEventListening`:
+
+<CodeBlock slots="heading, code" repeat="1" languages="Swift" />
+
+#### Swift
+
+```swift
+struct HomePage: View, ContentCardUIEventListening {
+    
+    @State var savedCards : [ContentCardUI] = []
+    
+    var body: some View {
+        ScrollView (.vertical) {
+          // Display the content cards here
+        }
+        .onAppear() {
+            let homePageSurface = Surface(path: "homepage")
+            // 2. Pass the listener to the getContentCardsUI API
+            AEPSwiftUI.getContentCardsUI(for: homePageSurface,
+                                         listener: self) { result in
+                switch result {
+                case .success(let cards):
+                    savedCards = cards
+                    
+                case .failure(let error):
+                    // handle error here                    
+                }
+            }
+        }
+    }
+    
+    // Implement the ContentCardUIEventListening protocol methods
+    func onDisplay(_ card: ContentCardUI) {
+        // Handle the card display event
+    }
+    
+    func onDismiss(_ card: ContentCardUI) {
+        // Handle the card dismiss event
+    }
+    
+    func onInteract(_ card: ContentCardUI, _ interactionId: String, actionURL: URL?) -> Bool {
+        // Handle the card interaction event
+        return false
+    }
+}
+```
+
+## Handling actionable URLs
+
+The `onInteract` method provides an optional `actionURL` parameter associated with the interaction event. The return value of this method determines how the URL is handled.
+
+- Return `true` if your application has successfully handled the URL. This indicates to the SDK that no further action is needed.
+
+- Return `false` to allow the SDK to process the URL.
```

Comment:
"Returns `true/false`...."

Reply:
Same comment as above, instructions are directed at the developer implementing the method.
Hence Return makes sense

Reply:
"Returns" would still be the correct usage here, even with its developer-focused and imperative tone since we're explaining that it will be returned and not directly providing a return statement. Here's a link to other documentation in the SDK where we use "returns" for the same usage case:

https://developer.adobe.com/client-sdks/edge/adobe-journey-optimizer/public-classes/content-card-schema-data/#getcontentcard---deprecated-1